sound/soc/sof/intel/hda-dsp.c

```diff
@@ -1437,12 +1490,19 @@ void hda_dsp_get_state(struct snd_sof_dev *sdev, const char *level)
 	else
 		module_text = fsr_module_names[module];
 
-	if (module == FSR_MOD_BRNGUP)
+	if (module == FSR_MOD_BRNGUP) {
 		state_text = hda_dsp_get_state_text(state, fsr_bringup_state_names,
 						    ARRAY_SIZE(fsr_bringup_state_names));
-	else
-		state_text = hda_dsp_get_state_text(state, fsr_rom_state_names,
-						    ARRAY_SIZE(fsr_rom_state_names));
+	} else {
+		if (chip->hw_ip_version < SOF_INTEL_ACE_1_0)
+			state_text = hda_dsp_get_state_text(state,
```

Comment:
I have an idea that we build a array for each ip version and then we use  fsr_rom_state_names[chip->hw_ip_version]

Reply:
so far we have two version, one for <= CAVS_2_5 and one for >= ACE_1_0.
At this point I would not bother with that, but as soon as we need another state table we should do that, I agree.